diff -urN gcc-12.2.0-diff/libsanitizer/asan/asan_linux.cpp gcc-12.2.0/libsanitizer/asan/asan_linux.cpp
--- gcc-12.2.0-diff/libsanitizer/asan/asan_linux.cpp	2022-09-21 22:59:44.758432852 +0800
+++ gcc-12.2.0/libsanitizer/asan/asan_linux.cpp	2022-09-21 22:58:51.075097959 +0800
@@ -65,6 +65,10 @@
 #define ucontext_t xucontext_t
 #endif
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 typedef enum {
   ASAN_RT_VERSION_UNDEFINED = 0,
   ASAN_RT_VERSION_DYNAMIC,
